src/tag_propagation.cpp

```diff
@@ -0,0 +1,97 @@
+#include "tag_propagation.h"
+
+#include <algorithm>
+#include <cassert>
+#include <cstddef>
+#include <sstream>
+#include <stdexcept>
+
+namespace datadog {
+namespace opentracing {
+
+const ot::string_view trace_tag_prefix = "_dd.p.";
+
+// The following [eBNF][1] grammar describes the tag propagation encoding.
+// The grammar was copied from [an internal design document][2].
+//
+//     tagset = ( tag, { ",", tag } ) | "";
+//     tag = ( identifier - space ), "=", identifier;
+//     identifier = allowed characters, { allowed characters };
+//     allowed characters = ( ? ASCII characters 32-126 ? - equal or comma );
+//     equal or comma = "=" | ",";
+//     space = " ";
+//
+// That is, comma-separated "<key>=<value>" pairs.
+//
+// See `tag_propagation_test.cpp` for examples.
+//
+// [1]: https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form
+// [2]:
+// https://docs.google.com/document/d/1zeO6LGnvxk5XweObHAwJbK3SfK23z7jQzp7ozWJTa2A/edit#heading=h.yp07yuixga36
```

Comment:
Caleb: Probably better not to reference internal docs.  If there's context you want to refer to, consider putting it in the repo itself. Alternatively, omit that context.